client/core/core.go

```diff
@@ -1373,79 +1374,100 @@ func (c *Core) initializeDEXConnections(crypter encrypt.Crypter) []*DEXBrief {
+				c.notify(newFeePaymentNote("Account unlock error", details, db.ErrorLevel, dc.acct.host))
+				return
 			}
-			c.reFee(dcrWallet, dc)
-			continue
-		}
-
-		wg.Add(1)
-		go func(dc *dexConnection) {
-			defer wg.Done()
-			err := c.authDEX(dc)
+			acctInfo, err := c.db.Account(dc.acct.id)
 			if err != nil {
-				details := fmt.Sprintf("%s: %v", dc.acct.host, err)
+				log.Errorf("Error retrieving account: %v", err)
+				return
+			}
+			if acctInfo.Disabled {
+				return
+			}
+			authDEXErr := c.authDEX(dc)
+			var mErr *msgjson.Error
+			// Disable account on all authentication errors, including when the server account is
+			// marked as unpaid while the client account is marked as paid
+			if errors.As(authDEXErr, &mErr) &&
+				(mErr.Code != msgjson.UnpaidAccountError ||
+					(mErr.Code == msgjson.UnpaidAccountError && dc.acct.feePaid())) {
+				details := fmt.Sprintf("%s: %v", dc.acct.host, authDEXErr)
 				c.notify(newFeePaymentNote("DEX auth error", details, db.ErrorLevel, dc.acct.host))
 				result.AuthErr = details
+				err = c.db.DisableAccount(acctInfo)
+				if err != nil {
+					log.Errorf("Error disabling account: %v", err)
+				}
 				return
 			}
 			result.Authed = true
-		}(dc)
+			enabledDEXConnections[host] = dc
+			// The user is authed if the account is successfully authenticated with the dex.
+			// Even unpaid accounts are authenticated, but with limited exchange capabilities.
+			// Now do some house cleaning in regard to registration fee payment.
+			if dc.acct.authed() {
+				return
+			}
+			if !dc.acct.feePaid() {
+				if len(dc.acct.feeCoin) == 0 {
+					details := fmt.Sprintf("Empty fee coin for %s.", dc.acct.host)
+					c.notify(newFeePaymentNote("Fee coin error", details, db.ErrorLevel, dc.acct.host))
+					result.AuthErr += details
+					return
+				}
+				// Try to unlock the Decred wallet, which should run the reFee cycle, and
+				// in turn will run authDEX.
+				dcrID, _ := dex.BipSymbolID("dcr")
+				dcrWallet, err := c.connectedWallet(dcrID)
+				if err != nil {
+					log.Debugf("Failed to connect for reFee at %s with error: %v", dc.acct.host, err)
+					details := fmt.Sprintf("Incomplete registration detected for %s, but failed to connect to the Decred wallet", dc.acct.host)
+					c.notify(newFeePaymentNote("Wallet connection warning", details, db.WarningLevel, dc.acct.host))
+					result.AuthErr += details
+					return
+				}
+				if !dcrWallet.unlocked() {
+					err = unlockWallet(dcrWallet, crypter)
+					if err != nil {
+						details := fmt.Sprintf("Connected to Decred wallet to complete registration at %s, but failed to unlock: %v", dc.acct.host, err)
+						c.notify(newFeePaymentNote("Wallet unlock error", details, db.ErrorLevel, dc.acct.host))
+						result.AuthErr += details
+						return
+					}
+				}
+				c.reFee(dcrWallet, dc)
+				return
+			}
+		}(host, dc)
 	}
 	wg.Wait()
+	c.connMtx.RUnlock()
+	c.connMtx.Lock()
```

Comment:
I got into trouble using write lock for the whole thing because --> connectedWallet() --> connected() issues a read lock. That seemed to lock things up entirely.

I'll see if I can resolve this by looking deeper into what all really needs to happen in parallel per your "What's the motivation for expanding the goroutines cover these operations in parallel now?" question.